src/content/en/fundamentals/native-hardware/recording-video/index.md

```diff
@@ -0,0 +1,212 @@
+attaching an `onchange` event to the input element and then reading
+the `files` property of the event object.
+
+<pre class="prettyprint">
+&lt;input type="file" accept="video/*" capture="camera" id="recorder">
+&lt;video id="player" controls>&lt;/video>
+&lt;script>
+  var recorder = document.getElementById('recorder');
+  var player = document.getElementById('player')'
+
+  recorder.addEventListener('change', function(e) {
+    var file = e.target.files[0];
+    // Do something with the video file.
+    player.src = URL.createObjectURL(file);
+  });
+&lt;/script>
+</pre>
+
+Once you have access to the file you can do anything you want with it. For
+example, you can:
+
+* Attach it directly to a `<video>` element so that you can play it
+* Download it to the user's device
+* Upload it to a server by attaching to an `XMLHttpRequest`
+* Draw the frames into a canvas and apply filters to it
+
+Whilst using the input element method of getting access to video data is
+ubiquitous, it is the least appealing option. We really want to get access to
+the camera and provide a nice experience directly in the page.
+
+## Access the camera interactively
+
+Modern browsers can have a direct line to the camera allowing us to build
+experiences that are fully integrated with the web page and the user will never
+leave the browser.
+
+### Acquire access to the camera
+
+We can directly access the camera by using an API in the WebRTC
+specification called `getUserMedia()`. `getUserMedia()` will prompt the user for
+access to their connected microphones and cameras.
+
+If successful the API will return a `Stream` that will contain the data from
+either the camera or the microphone, and we can then either attach it to
+a `<video>` element or save it using the `MediaRecorder` API.
+
+To get data from the camera we just set `video: true` in the constraints
+object that is passed to the `getUserMedia()` API
+
+<pre class="prettyprint">
+&lt;video id="player" controls>&lt;/video>
+&lt;script>
+  var player = document.getElementById('player');
+
+  var handleSuccess = function(stream) {
+    player.srcObject = stream;
+  };
+
+  navigator.mediaDevices.getUserMedia({ audio: true, video: true })
+      .then(handleSuccess)
+&lt;/script>
+</pre>
+
+By itself, this isn't that useful. All we can do is take the video data and play
+it back.
+
+### Access the raw data from the camera
+
+To access the raw video data from the camera you can draw each frame into a `<canvas>` and
+manipulate the pixels directly. You can learn more about this in our article about
+[applying real-time effects to images and video](/web/fundamentals/media/manipulating/live-effects).
+
+### Save the data from the camera
+
+The easiest way to save the data from the camera is to use the
+`MediaRecorder` API.
+
+The `MediaRecorder` API will take the stream created by `getUserMedia` and then
+progressively save the data from the stream to you preferred destination.
+
+<pre class="prettyprint">
```

Comment:
I didn't think we wrapped code in pre and pretty print?

@petele ?

Reply:
The WF linter complains if you have a bare `<script>` tag in the article. This is the way that I have seen others getting around this, but open to alternatives.

Reply:
The `<script>` tag warning is just that, meant to catch an accidental script tag, if you know you're in a code block, it can be ignored.